Sources/Metrics/Metrics.swift

```diff
@@ -74,3 +74,20 @@ extension Timer {
         }
     }
 }
+
+#if swift(>=5.7)
+extension Timer {
+    /// Convenience for recording a duration based on ``Duration``.
+    ///
+    /// - parameters:
+    ///     - duration: The duration to record.
+    @available(macOS 13, iOS 16, tvOS 15, watchOS 8, *)
+    @inlinable
+    public func record(_ duration: Duration) {
+        // `Duration` doesn't have a nice way to convert it back to `Double` of seconds,
+        // so instead we can multiply attoseconds by 1e18 and add the number of seconds to it.
+        let durationSeconds = Double(duration.components.seconds) + Double(duration.components.attoseconds) / 1e18
+        self.recordSeconds(durationSeconds)
```

Comment:
Actually, this is _too_ lossy, we have `recordNanoseconds` that this should call into, as it's the most precision we have. With just seconds this would not be very useful for fast operations